ethcore/src/builtin.rs

```diff
@@ -273,10 +324,13 @@ impl Impl for Ripemd160 {
 fn modexp(mut base: BigUint, mut exp: BigUint, modulus: BigUint) -> BigUint {
 	use num::Integer;
 
+	if modulus <= BigUint::one() {
+		return BigUint::zero();
+	}
```

Comment:
to clarify, this is expected behavior, applied in order?
 `n^m % 0` || `n^m % 1` -> 0
  `n^0 % m` -> 1
  `0^n % m` -> 0

I think it would still be more clear in the match statement, just putting the modulus arm at the top.

Reply:
After some discussion we've decided to split it into `if` statements instead.